substrate/utils/frame/benchmarking-cli/src/pallet/command.rs

```diff
@@ -589,13 +578,34 @@ impl PalletCmd {
 					.build_storage()
 					.map_err(|e| format!("{ERROR_CANNOT_BUILD_GENESIS}\nError: {e}"))?;
 
-				(storage, Default::default())
+				storage
 			},
 			(Some(GenesisBuilder::Runtime), _) | (None, true) =>
-				(Default::default(), self.genesis_from_runtime::<H, F>()?),
+				self.genesis_from_runtime::<H, F>().and_then(Self::changes_to_storage)?,
 		})
 	}
 
+	/// Convert some overlayed changes to a storage.
+	fn changes_to_storage<H: Hash>(
```

Comment:
Why not get rid of all of this and `genesis_from_runtime` and just use `GenesisConfigBuilderRuntimeCaller`? It has all the machinery ready and we get rid of the manual json merging stuff too.

Reply:
I'd also try using `GenesisConfigBuilderRuntimeCaller` to have genesis storage building in one place. If something is missing it can be adjusted / reworked.

Reply:
I changed it now.